src/workerd/jsg/resource.h

```diff
@@ -191,6 +192,125 @@ template <typename TypeWrapper,
     typename Indexes>
 struct MethodCallback;
 
+// Helper to determine if a type can be used as a return type in V8 Fast API
+template <typename T>
+struct IsFastApiCompatibleReturnType {
+  static constexpr bool value = std::is_same_v<T, void> || std::is_same_v<T, bool> ||
+      std::is_same_v<T, int32_t> || std::is_same_v<T, uint32_t> || std::is_same_v<T, float> ||
+      std::is_same_v<T, double>;
+};
+
+// Helper to determine if a type can be used as a parameter in V8 Fast API
+template <typename T>
+struct IsFastApiCompatibleParamType {
+  static constexpr bool value = IsFastApiCompatibleReturnType<T>::value ||
+      std::is_same_v<T, v8::Local<v8::Value>> || std::is_same_v<T, v8::Local<v8::Object>>;
+};
+
+// Helper to determine if a method can be used with V8 fast API
+template <typename Ret, typename... Args>
+struct IsFastApiCompatible {
+  static constexpr bool value = IsFastApiCompatibleReturnType<Ret>::value &&
+      (IsFastApiCompatibleParamType<Args>::value && ...);
+};
+
+// Helper to determine if a method pointer type is compatible with Fast API
+template <typename Method>
+struct IsFastMethodCompatible {
+  static constexpr bool value = false;
+};
+
+// Specialization for non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...)> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
+
+// Specialization for const non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...) const> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
```

Comment:
Nit: It would be helpful to have some `static_assert` tests in `fast-method-test.c++` that verify these work as expected for method signatures that are *not* fast-method compatible.

Reply:
Last commit extends these tests. Thank you for the review.